### ISSUE TITLE
AV sync  after SDI input switching

### ISSUE THREAD

**Issue by yverbin** [lu-zero/bmdtools#75]
Audio and video out of sync after SDI input switching. I try to use DeckLink Mini Recorder, DeckLink SDI - every SDI input switching  increases  delay between audio and video streams.

**Comment by stevenviola** [comment]
I've seen the same behavior on the Decklink SDI (pre 4k) and Mini Recorder. This seems to be an issue with the firmware on the cards and wasn't something wrong with bmdtools. I created posts on the blackmagic forums regarding this for the [Decklink SDI](https://forum.blackmagicdesign.com/viewtopic.php?f=12&t=39143) and the [Decklink Mini Recorder](https://forum.blackmagicdesign.com/viewtopic.php?f=12&t=44316) with more details

Basically, the pts values of the audio and video differ a lot after an input switch which causes A/V sync issue. These pts values come from the card and a lot of issues were resolved for me with the Decklink SDI card by updating the firmware and drivers to 10.4.2. I've not tested the 10.6.X releases yet, but those might fix other issues with the Mini Recorder as well

**Comment by yverbin** [comment]
@stevenviola In some cases, I use open broadcaster for capture & encode, and all looks ok.

**Comment by yverbin** [comment]
I successfully modified decklink_input_callback::VideoInputFrameArrived method of ffmpeg decklink decoding module for ts synchronization a/v streams after  SDI input switching, and at first glance, all looks ok. Now i'm testing it

**Comment by lu-zero** [comment]
@yverbin glad you have something working, but this does not belong here.

**Comment by ggnull35** [comment]
@yverbin can you share your update so we can review it as well ? Thank you.
